lib/iris/analysis/_regrid.py

```diff
@@ -543,3 +544,336 @@ def __call__(self, src):
+        projected_src_points = projection.transform_points(
+            src_projection, src_x_coord.points, src_y_coord.points)
+
+        tgt_x, tgt_y = np.meshgrid(tgt_x_coord.points, tgt_y_coord.points)
+        tgt_projection = grid_x_coord.coord_system.as_cartopy_projection()
+        projected_tgt_grid = projection.transform_points(
+                tgt_projection, tgt_x, tgt_y)
+
+        # Prepare the result data array.
+        tgt_y_shape, = tgt_y_coord.shape
+        tgt_x_shape, = tgt_x_coord.shape
+        tgt_shape = src_cube.shape[:xy_dim] + tgt_y_shape + tgt_y_shape + src_cube.shape[xy_dim+1:]
+        # XXX TODO: Deal with masked src_data
+        data = np.empty(tgt_shape, dtype=src_data.dtype)
+
+        iter_shape = list(src_data.shape)
+        iter_shape[xy_dim] = 1
+
+        for index in np.ndindex(tuple(iter_shape)):
+            src_index = list(index)
+            src_index[xy_dim] = slice(None)
+            src_subset = src_data[tuple(index)]
+            tgt_index = index[:xy_dim] + (slice(None), slice(None)) + index[xy_dim+1:]
+            data[tgt_index] = griddata(projected_src_points[...,:2],
+                                       src_subset,
+                                       projected_tgt_grid[...,:2],
+                                       method=method)
+        return data
+
+    @staticmethod
+    def _create_cube(data, src_cube, xy_dim, src_x_coord, src_y_coord,
+                     grid_x_coord, grid_y_coord, regrid_callback):
+        """
+        Return a new Cube for the result of regridding the source Cube onto
+        the new grid.
+
+        All the metadata and coordinates of the result Cube are copied from
+        the source Cube, with two exceptions:
+            - Grid dimension coordinates are copied from the grid Cube.
+            - Auxiliary coordinates which span the grid dimensions are
+              ignored, except where they provide a reference surface for an
+              :class:`iris.aux_factory.AuxCoordFactory`.
+
+        Args:
+
+        * data:
+            The regridded data as an N-dimensional NumPy array.
+        * src_cube:
+            The source Cube.
+        * xy_dim:
+            The dimension within the source Cube that the x and y coordinate
+            span.
+        * src_x_coord:
+            The X :class:`iris.coords.DimCoord`.
+        * src_y_coord:
+            The Y :class:`iris.coords.DimCoord`.
+        * grid_x_coord:
+            The :class:`iris.coords.DimCoord` for the new grid's X
+            coordinate.
+        * grid_y_coord:
+            The :class:`iris.coords.DimCoord` for the new grid's Y
+            coordinate.
+        * regrid_callback:
+            The routine that will be used to calculate the interpolated
+            values of any reference surfaces.
+
+        Returns:
+            The new, regridded Cube.
+
+        """
+        #
+        # XXX: At the moment requires to be a static method as used by
```

Comment:
it would be helpful to understand why this is used by the experimental code. It's not clear to me why this should be a static method